stdlib/string.ml

```diff
@@ -105,14 +105,10 @@ let trim s =
   else s
 
 let escaped s =
-  let rec escape_if_needed s n i =
-    if i >= n then s else
-      match unsafe_get s i with
-      | '\"' | '\\' | '\000'..'\031' | '\127'.. '\255' ->
-          bts (B.escaped (bos s))
-      | _ -> escape_if_needed s n (i+1)
-  in
-  escape_if_needed s (length s) 0
+  let b = bos s in
+  (* We satisfy [unsafe_escape]'s precondition by passing an
+     immutable byte sequence. *)
+  bts (B.unsafe_escape ~share:true b)
```

Comment:
Why a `let`?
```
  bts (B.unsafe_escape (bos s))
```
or, more modern
```
  s |> bos |> B.unsafe_escape |> bts
```

Reply:
It's purely stylistic, but the comment refers to "an immutable byte sequence" which is `b`. I prefer to refer to a named value defined right above than to a subexpression of a non-trivial computation.